examples/mixture/plot_gmm_init.py

```diff
@@ -0,0 +1,113 @@
+"""
+==========================
+GMM Initialization Methods
+==========================
+
+Examples of the different methods of initialization in Gaussian Mixture Models
+
+See :ref:`gmm` for more information on the estimator.
+
+Here we generate some sample data with four easy to identify clusters. The
+purpose of this example is to show the four different methods for the
+initialization parameter *init_param*.
+
+The four initializations are *kmeans* (default), *random*, *rand_data* and
+*k-means++*.
+
+Orange diamonds represent the initialization centers for the gmm generated by
+the *init_param*. The rest of the data is represented as crosses and the
+colouring represents the eventual associated classification after the GMM has
+finished.
+
+The numbers in the top right of each subplot represent the number of
+iterations taken for the GaussianMixture to converge and the relative time
+taken for the initialization part of the algorithm to run. The shorter
+initialization times tend to have a greater number of iterations to converge.
+
+The initialization time is the ratio of the time taken for that method versus
+the time taken for the default *kmeans* method. As you can see all three
+alternative methods take less time to initialize when compared to *kmeans*.
+
+In this example, when initialized with *rand_data* or *random* the model takes
+more iterations to converge. Here *k-means++* does a good job of both low
+time to initialize and low number of GaussianMixture iterations to converge.
+"""
+
+
+# Author Gordon Walsh (github: g-walsh) gordon.p.walsh@gmail.com
+# data generation code from https://jakevdp.github.io/
+
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.mixture import GaussianMixture
+from sklearn.utils.extmath import row_norms
+from sklearn.datasets.samples_generator import make_blobs
+from timeit import default_timer as timer
+
+print(__doc__)
+
+# Generate some data
+
+X, y_true = make_blobs(n_samples=4000, centers=4,
+                       cluster_std=0.60, random_state=0)
+X = X[:, ::-1]
+
+n_samples = 4000
+n_components = 4
+x_squared_norms = row_norms(X, squared=True)
+
+
+def gen_gmm(init_params, seed=None):
+    r = np.random.RandomState(seed)
+    start = timer()
+    init_means = calculate_means(X, init_params, r)
+
+    end = timer()
+    init_time = end - start
+
+    gmm = GaussianMixture(n_components=4, means_init=init_means, tol=1e-9,
+                          max_iter=2000, random_state=r).fit(X)
+
+    labels = gmm.predict(X)
+    iterations = gmm.n_iter_
+    return labels, init_means, seed, init_params, init_time, iterations
+
+
+def calculate_means(X, init_params, r):
+    gmm = GaussianMixture(n_components=4, init_params=init_params, tol=1e-9,
```

Comment:
Maybe you should leave a comment here that max_iter=0 allows you to see the effect of initialisation